plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -504,6 +511,56 @@ public void testDeletingEntirePartitionedTable()
         assertThat(this.loadTable(tableName).newScan().planFiles()).hasSize(0);
     }
 
+    @Test
+    public void testFilesTable()
+            throws Exception
+    {
+        String tableName = "test_files_table_" + randomNameSuffix();
+        String tableLocation = metastoreDir.getPath() + "/" + tableName;
+        assertUpdate("CREATE TABLE " + tableName + " WITH (location = '" + tableLocation + "', format_version = 2) AS SELECT * FROM tpch.tiny.nation", 25);
+        BaseTable table = loadTable(tableName);
+        Metrics metrics = new Metrics(
+                10L,
+                ImmutableMap.of(1, 2L, 2, 3L),
+                ImmutableMap.of(1, 5L, 2, 3L, 3, 2L),
+                ImmutableMap.of(1, 0L, 2, 2L),
+                ImmutableMap.of(4, 1L),
+                ImmutableMap.of(1, ByteBuffer.allocate(8).order(ByteOrder.LITTLE_ENDIAN).putLong(0, 0L)),
+                ImmutableMap.of(1, ByteBuffer.allocate(8).order(ByteOrder.LITTLE_ENDIAN).putLong(0, 4L)));
+        DataFile dataFile =
+                DataFiles.builder(PartitionSpec.unpartitioned())
+                        .withFormat(FileFormat.ORC)
+                        .withPath(tableLocation + "/data/20230310_133318_00004_qhftg-870c1a85-e56c-4536-9b84-33c69166ddfb.orc")
+                        .withFileSizeInBytes(1234)
+                        .withMetrics(metrics)
+                        .withSplitOffsets(ImmutableList.of(4L))
+                        .withEncryptionKeyMetadata(ByteBuffer.wrap("Trino".getBytes(UTF_8)))
+                        .build();
+        table.newAppend().appendFile(dataFile).commit();
+        writeEqualityDeleteToNationTable(table);
+        assertQuery(
```

Comment:
Could not include `file_path` and `file_size_in_bytes`, As their values will be random.

Reply:
Those values are already covered in `BaseIcebergConnectorTest#testFileSizeInManifest`. No need to verify them in this test, but worth to leave a code comment in my opinion.

Reply:
I used `nation` table because it was easy to construct and equality delete can be applied easily (as we already have `writeEqualityDeleteToNationTable`). So Ill leave a comment.